.circleci/config.yml

```diff
@@ -253,6 +256,561 @@ commands:
+  cf_backup:
+    description: "Login to cloud foundry space with service account credentials, Connect to DB & S3, backup DB to S3"
+    parameters:
+      app_name:
+        description: "Name of Cloud Foundry cloud.gov application; must match
+          application name specified in manifest"
+        type: string
+      auth_client_id:
+        description: "Name of CircleCi project environment variable that
+          holds authentication client id, a required application variable"
+        type: env_var_name
+      auth_client_secret:
+        description: "Name of CircleCi project environment variable that
+          holds authentication client secret, a required application variable"
+        type: env_var_name
+      cloudgov_username:
+        description: "Name of CircleCi project environment variable that
+          holds deployer username for cloudgov space"
+        type: env_var_name
+      cloudgov_password:
+        description: "Name of CircleCi project environment variable that
+          holds deployer password for cloudgov space"
+        type: env_var_name
+      cloudgov_space:
+        description: "Name of CircleCi project environment variable that
+          holds name of cloudgov space to target for application deployment"
+        type: env_var_name
+    steps:
+      - run:
+          name: Install Dependencies
+          command: |
+            # update
+            sudo apt-get update
+            # Install PostgreSQL Client to have access to pg_dump
+            sudo apt-get install -y postgresql-client
+            # Install jq
+            sudo apt-get install -y jq
+            # Install zip & unzip
+            sudo apt-get install -y zip unzip
+            # Install Curl
+            sudo apt-get install -y curl
+            # Install uuid-runtime to have access to uuidgen
+            sudo apt-get install uuid-runtime
+            # Install pv
+            sudo apt-get install pv
+            # Install wget
+            sudo apt-get install wget
+            # Install Cloud Foundry CLI
+            wget -q -O - https://packages.cloudfoundry.org/debian/cli.cloudfoundry.org.key | sudo apt-key add -
+            echo "deb https://packages.cloudfoundry.org/debian stable main" | sudo tee /etc/apt/sources.list.d/cloudfoundry-cli.list
+            sudo apt-get update
+            sudo apt-get install cf8-cli
+            # cleanup
+            sudo rm -rf /var/lib/apt/lists/*
+      - run:
+          name: Login with service account
+          command: |
+            cf login -a << pipeline.parameters.cg_api >> \
+              -u ${<< parameters.cloudgov_username >>} \
+              -p ${<< parameters.cloudgov_password >>} \
+              -o << pipeline.parameters.cg_org >> \
+              -s ${<< parameters.cloudgov_space >>}
+      - run:
+          name: Connect to DB
+          command: |
+            # Create the script file
+            cat \<<<'EOF' > /tmp/setup_db_connection.sh
```

Comment:
Can these scripts be checked into the repo & called via a makefile/ taskfile?

Will make this ci/cd config a bit more manageable.

Reply:
if i keep them inline then no checkout is needed speeding up the process